### PR TITLE
Adds defining target architecture post

### PR DESCRIPTION
**Pull request by andrewabest**
# About

This is a post about defining target architecture, which is something we did within the engineering team to help inform our 2023/24 technical strategy.

# Release

No, this is an "engineering general interest" post, and isn't associated with a particular release.

# Blog image idea

Perhaps riff on the image used in https://octopus.com/blog/creating-new-architecture ?

# Prerequisites

- [x] The draft is complete and this post is ready to be reviewed.
- [x] I'm familiar with the [guide for blog content basics](https://www.octopus.design/932c0f1a9/p/901d2a-blog-content) and [writing educational content](https://www.octopus.design/932c0f1a9/p/137b06-educational-content---how-we-write-about-what-we-do).
- [x] I've checked the content using Grammarly (Octopus has a corporate license) and the [Hemingway app](https://hemingwayapp.com/) to remove typos, grammatical errors, and passive voice. 
- [x] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://www.octopus.design/932c0f1a9/p/901d2a-blog-content/t/817249) and none of those apply.
- [x] The images I've included follow the [rules for working with images](https://www.octopus.design/932c0f1a9/p/5061d7-working-with-images).
- [x] I've added alt text for all the images I've included (alt text describes an image to people unable to see it, 125 characters max)
- [ ] I've drafted social media posts for Twitter and LinkedIn and will post them for review in #topic-social-media.
- [x] I've dropped a note in Slack in #topic-content-marketing to let Tegan Ali know my post is ready for review. (No need to fill out a content brief.)

## How to review this PR

If there's anything you'd like reviewers to focus on, mention it here.

## Publication date

I'm happy to ship this once it has been reviewed and meets standards.

### CODE REVIEW

**Comment by andrewabest** [comment]
I need to comb through this to adhere to the prerequisites, so do not review for now.

**Comment by andrewabest** [comment]
@TeganAli it is a bit unclear to me whether I should wait to know when this blog post will be posted before posting to #topic-social-media. Or do I just need to give them a snippet and they will coordinate with you as to the release date?

**Comment by TeganAli** [comment]
@andrewabest probably best to wait, in case any of the edits impact what you'd write too. Thanks :)

**Comment by TeganAli** [comment]
Awesome post, @andrewabest . Really interesting read. I'll do another pass before publishing to make sure I haven't introduced any errors. Please let me know if you have any questions or disagree with any edits :)

File: blog/2023-11/defining-target-architecture/index.md [hunk 1]
```diff
@@ -12,23 +12,23 @@ tags:
   - Engineering
 ---
 
-When writing a technical strategy, having a clear picture of the technical problems you are facing in your organization is useful.
+When writing a technical strategy, it's useful to have a clear picture of the technical problems you're facing in your organization.
 
-Ideally, you should be able to make a clear problem diagnosis based on knowing where the problems you are facing are going to impact your organizations business strategy, and at what magnitude.
+Ideally, you should be able to diagnose a problem clearly, knowing how the problems are going to impact your organization's business strategy, and at what magnitude.
```
Comment (andrewabest) [inline blog/2023-11/defining-target-architecture/index.md:17]: This edit has diluted this paragraph a bit. I'd like it to still talk about "A clear diagnosis" rather than "diagnosing a problem clearly", as the diagnosis is a key part of good strategy. 

I'd suggest this as an alternative edit:

> Ideally, you should be able to make a clear problem diagnosis, knowing which technical problems are going to impact your organizations business strategy, and at what magnitude.

File: blog/2023-11/defining-target-architecture/index.md [hunk 2]
```diff
@@ -12,23 +12,23 @@ tags:
...
 
-A common state that many engineering organizations end up in is where each team in the org has its own perspective on what _the biggest problems_ are. Often they are the problems that have an outsized impact on that particular team.
+A common state that many engineering organizations end up in is where each team has its own perspective on what _the biggest problems_ are. Often they're the problems that have an outsized impact on that particular team.
 
-Occasionally, there are shared problems that all teams can agree on. This is more likely if teams use shared infrastructure or work within a monorepo or monolithic application. Even if the teams agree on the problem, they may have different perspectives on its impact, particularly regarding how much it impacts the business's strategy.
+Occasionally, you'll find shared problems that all teams can agree on. This is more likely if teams use shared infrastructure or work in a mono-repo or monolithic application. Even if the teams agree on the problem, they may have different perspectives on its impact. In particular, how much it impacts the business's strategy.
```
Comment (andrewabest) [inline blog/2023-11/defining-target-architecture/index.md:21]: Can we unhyphenate `monorepo` here please? It isn't hyphenated in most technical literature.

Comment (andrewabest) [reply blog/2023-11/defining-target-architecture/index.md:21]: Also if it has been hyphenated in other edits :)

File: blog/2023-11/defining-target-architecture/index.md [hunk 3]
```diff
@@ -108,27 +112,38 @@ As we dove into individual capabilities, we quickly found that:
 - Spending time writing about things that we marked as high maturity felt low value
 - Fully exploring areas that we'd marked as low maturity took _a lot of time_
 
-The combination of these two insights led to us removing everything marked as _high maturity_ from the document so that we might focus our time and effort on the problems that needed to be solved.
+The combination of these two insights led to us removing everything marked as _high maturity_ from the document. This let us focus our time and effort on the problems we needed tosolved.
```
Comment (andrewabest) [inline blog/2023-11/defining-target-architecture/index.md]: Typo here - `tosolved` should be `to solve`.

File: blog/2023-11/defining-target-architecture/index.md [hunk 4]
```diff
@@ -108,27 +112,38 @@ As we dove into individual capabilities, we quickly found that:
...
+It's worth highlighting the impact of this choice. Instead of the target architecture painting a full picture of the architecture underpinning our product, it focused on the problems that needed to be solved. This was a good tradeoff. By focusing on the problems that needed solving, we could generate a stronger signal and reduce the noise from communicating the detail of capabilities that were already fit for purpose.
+:::
 
 ### Impact and effort
 
-Once our target state exploration was completed, we had a set of well-defined problems and solution preferences underneath each capability.
+After we completed our target state exploration, we had a set of well-defined problems and solution preferences underneath each capability.
 
-It became quickly apparent that the documented target state was hard to reason about:
+It became clear that the documented target state was hard to reason about:
```
Comment (andrewabest) [inline blog/2023-11/defining-target-architecture/index.md:125]: This reads a little awkwardly - could we rephrase it to 

> At this point it became clear that the documented target state was hard to reason about

File: blog/2023-11/defining-target-architecture/index.md [hunk 5]
```diff
@@ -147,70 +147,70 @@ When determining effort, experts reasoned about the relative efforts of particul
...
-This is the secret to unlocking scale in engineering. As much as possible, you want to remove any key-role dependencies and enable teams and individuals to reason autonomously, leveraging clearly communicated frames and goals.
+This is the secret to unlocking scale in engineering. As much as possible, remove any key-role dependencies. Let teams and individuals reason autonomously, using clearly communicated frames and goals.
 
 ### Technical education
 
-Beyond technical people being empowered with the knowledge of target architectural state, similar impacts are realised for non-technical stakeholders.
+Beyond empowering technical people with the knowledge of target architectural state, non-technical stakeholders realize similar impacts.
 
-Product, sales, support, marketing - people across the organization now have a way to understand what technical challenges need to be solved to achieve business goals.
+Product, sales, support, marketing - people across the organization can understand what technical challenges to solve to achieve business goals.
```
Comment (andrewabest) [inline blog/2023-11/defining-target-architecture/index.md:196]: This sentence is not quite right now - non technical people do not need to solve the technical challenges, they need to know which challenges need to be solved. Can we reword to:

> Product, sales, support, marketing - people across the organization can understand what technical challenges need to be solved to achieve business goals.

File: blog/2023-11/defining-target-architecture/index.md [hunk 6]
```diff
@@ -147,70 +147,70 @@ When determining effort, experts reasoned about the relative efforts of particul
...
 
-If you work in a company that produces a software product, or your business operations are largely powered by software, this understanding is critical. Often technical problems are some of the primary impediments to success - hard constraints that limit what the business can achieve.
+If you work in an organization that produces a software product, or your business operations are largely powered by software, this understanding is critical. Often technical problems are some of the primary impediments to success. They're hard constraints that limit what the business can achieve.
 
-By aligning the organization with the technical problems that need to be solved and their impact on goals, good decisions can be made when allocating budget and investing in solving those problems.
+By aligning the organization with the technical problems you need to solve and their impact on goals, everyone can make good decisions when allocating budget and investing in solving those problems.
```
Comment (andrewabest) [inline blog/2023-11/defining-target-architecture/index.md:200]: This paragraph isn't quite right. I'm not sure on empahsizing "you" here - they are problems the org needs to solve. And not everyone makes budgeting and investment decisions. 

I'd suggest:

By aligning the organization on technical problems and their impacts, decision makers can make informed decisions when allocating budget and investing in solving those problems.

**Comment by andrewabest** [comment]
@TeganAli I've left feedback on the edits - if you could either apply the feedback as-is, or give me feedback on my feedback so we can reach consensus on each, that would be grand :)

File: blog/2023-11/defining-target-architecture/index.md (hunk 3 above)
Comment (TeganAli) [reply blog/2023-11/defining-target-architecture/index.md]: Good catch :)

File: blog/2023-11/defining-target-architecture/index.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+bannerImage:
+bannerImageAlt: 125 characters max, describes image to people unable to see it.
+isFeatured: false
+tags:
+  - Engineering
+---
+
+When writing a technical strategy, it's useful to have a clear picture of the technical problems you're facing in your organization.
+
+Ideally, you should be able to diagnose a problem clearly, knowing how the problems are going to impact your organization's business strategy, and at what magnitude.
```
Comment (TeganAli) [inline blog/2023-11/defining-target-architecture/index.md:17]:
```suggestion
Ideally, you should be able to make a clear problem diagnosis, knowing which technical problems are going to impact your organization's business strategy, and at what magnitude.
```

File: blog/2023-11/defining-target-architecture/index.md (hunk 2 above)
Comment (TeganAli) [reply blog/2023-11/defining-target-architecture/index.md:21]: Sure thing

File: blog/2023-11/defining-target-architecture/index.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+A common state that many engineering organizations end up in is where each team has its own perspective on what _the biggest problems_ are. Often they're the problems that have an outsized impact on that particular team.
+
+Occasionally, you'll find shared problems that all teams can agree on. This is more likely if teams use shared infrastructure or work in a mono-repo or monolithic application. Even if the teams agree on the problem, they may have different perspectives on its impact. In particular, how much it impacts the business's strategy.
```
Comment (TeganAli) [inline blog/2023-11/defining-target-architecture/index.md:21]:
```suggestion
Occasionally, you'll find shared problems that all teams can agree on. This is more likely if teams use shared infrastructure or work in a monorepo or monolithic application. Even if the teams agree on the problem, they may have different perspectives on its impact. In particular, how much it impacts the business's strategy.
```

File: blog/2023-11/defining-target-architecture/index.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+These principles helped contributors by providing a frame for their exploration. They also helped future consumers of the target architecture by providing them with the knowledge of that frame, so that they could empathize with how we formed the opinions we authored.
+
+### Focus on what matters
+
+As we dove into individual capabilities, we quickly found that:
+
+- Spending time writing about things that we marked as high maturity felt low value
+- Fully exploring areas that we'd marked as low maturity took _a lot of time_
+
+The combination of these two insights led to us removing everything marked as _high maturity_ from the document. This let us focus our time and effort on the problems we needed tosolved.
```
Comment (TeganAli) [inline blog/2023-11/defining-target-architecture/index.md]:
```suggestion
The combination of these two insights led to us removing everything marked as _high maturity_ from the document. This let us focus our time and effort on the problems we needed to solve.
```

Comment (TeganAli) [inline blog/2023-11/defining-target-architecture/index.md]:
```suggestion
The combination of these two insights led to us removing everything marked as _high maturity_ from the document. This let us focus our time and effort on the problems we needed to solve.
```

Comment (TeganAli) [inline blog/2023-11/defining-target-architecture/index.md]:
```suggestion
The combination of these two insights led to us removing everything marked as _high maturity_ from the document. This let us focus our time and effort on the problems we needed to solve.
```

File: blog/2023-11/defining-target-architecture/index.md [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+
+:::hint
+It's worth highlighting the impact of this choice. Instead of the target architecture painting a full picture of the architecture underpinning our product, it focused on the problems that needed to be solved. This was a good tradeoff. By focusing on the problems that needed solving, we could generate a stronger signal and reduce the noise from communicating the detail of capabilities that were already fit for purpose.
+:::
+
+### Impact and effort
+
+After we completed our target state exploration, we had a set of well-defined problems and solution preferences underneath each capability.
+
+It became clear that the documented target state was hard to reason about:
```
Comment (TeganAli) [inline blog/2023-11/defining-target-architecture/index.md:125]:
```suggestion
At this point it became clear that the documented target state was hard to reason about:
```

File: blog/2023-11/defining-target-architecture/index.md (hunk 5 above)
Comment (TeganAli) [reply blog/2023-11/defining-target-architecture/index.md:196]: Sure thing. I was trying to avoid passive language but makes more sense this way so will change as per your suggestion :)

File: blog/2023-11/defining-target-architecture/index.md (hunk 6 above)
Comment (TeganAli) [reply blog/2023-11/defining-target-architecture/index.md:200]: Sure thing, I'll update :)

**Comment by TeganAli** [comment]
@andrewabest all done, I've applied all your suggestions. Thank you :)

**Review by TeganAli — APPROVED** [review]
